src/services/transaction/TransactionMaterialService.spec.ts

```diff
@@ -9,15 +9,46 @@ const transactionMaterialService = new TransactionMaterialService(
 
 describe('TransactionMaterialService', () => {
 	describe('getTransactionMaterial', () => {
-		it('works when ApiPromise works (block 789629)', async () => {
+		it('Should return the scale encoded metadata when the `metadata` query param is `scale`', async () => {
 			expect(
 				sanitizeNumbers(
 					await transactionMaterialService.fetchTransactionMaterial(
 						blockHash789629,
-						false
+						'scale'
 					)
 				)
 			).toStrictEqual(response789629);
 		});
+
```

Comment:
Is it worth also haveing a test that regardless of `metadata`, if `noMeta` is set, nothing is returned?

Reply:
Yea, I agree. I was thinking about that, and that will have to be tested using the controller. We send the service a argument `metadata` which can either be a string denoting the type of metadata we are going to return, or a value of false, meaning no metadata (You rewrote this section above). Little nit pick of my original logic, but do you think we should seperate those types and have two arguments passed in `metadata: boolean`, and `metadatFormat: string`, instead of just `metadata: string | false`.

Reply:
Considering the rework of the logic above, I think it's okay to have the current format. But tests to check are definitely needed.

Reply:
I've no strong opinions on the input args really :) coming from the query params I guess things will always be a touch messy until they have been parsed into something nicer!